app/controllers/TimeController.scala

```diff
@@ -0,0 +1,175 @@
+  }
+
+  def getUserWithWorkingHours(users: List[User], startDate: Calendar, endDate: Calendar)(implicit request: AuthenticatedRequest[AnyContent]): List[Fox[JsObject]] = {
+    for {
+      user <- users
+    } yield {
+      getUserHours(user, startDate, endDate)
+    }
+  }
+
+  def getUserHours(user: User, startDate: Calendar, endDate: Calendar)(implicit request: AuthenticatedRequest[AnyContent]): Fox[JsObject] = {
+    for {
+      timeList <- TimeSpanDAO.findByUser(user, Some(startDate.getTimeInMillis), Some(endDate.getTimeInMillis))
+      timeListWithTask <- getOnlyTimeSpansWithTask(timeList)
+      boxJs <- Future.traverse(timeListWithTask)(timeWrites(_))
+    } yield {
+      val js = boxJs.flatten
+      if(js.nonEmpty)
+        js.head.value.get("time") match {
+          case Some(x) =>
+            if (isFormatedDurationZero(x.toString())) None
+            else Some(Json.obj(
+              "user" -> Json.toJson(user)(User.userCompactWrites),
+              "timelogs" -> Json.toJson(js)))
+          case _ => None
+        }
+      else
+        None
+    }
+  }
+
+  def getOnlyTimeSpansWithTask(l: List[TimeSpan])(implicit request: AuthenticatedRequest[AnyContent]): Future[List[TimeSpan]] = {
+    for {
+      list <- Fox.sequence(l.map(t => getTimeSpanOptionTask(t)))
+    } yield {
+      list.flatten.flatten
+    }
+  }
+
+  def getTimeSpanOptionTask(t: TimeSpan)(implicit request: AuthenticatedRequest[AnyContent]): Fox[Option[TimeSpan]] = {
+    t.annotation match {
+      case Some(annotationId) => for {
+        annotation <- AnnotationDAO.findOneById(annotationId)
+      } yield {
+        annotation._task match {
+          case Some(_) => Some(t)
+          case None => None
+        }
+      }
+      case None => Fox(Future(None))
+    }
+  }
+
+  def timeWrites(timeSpan: TimeSpan)(implicit request: AuthenticatedRequest[AnyContent]): Fox[JsObject] = {
+    for {
+      task <- getTaskForTimeSpan(timeSpan)
+      tasktype <- TaskTypeDAO.findOneById(task._taskType)
+      project <- ProjectDAO.findOneByName(task._project)
+    } yield {
+      Json.obj(
+        "time" -> formatDuration(timeSpan.time),
+        "timestamp" -> timeSpan.timestamp,
+        "annotation" -> timeSpan.annotation,
+        "_id" -> timeSpan._id.stringify,
+        "task_id" -> task.id,
+        "project_name" -> project.name,
+        "tasktype_id" -> tasktype.id,
+        "tasktype_summary" -> tasktype.summary)
+    }
+  }
+
+  def getTaskForTimeSpan(timeSpan: TimeSpan)(implicit request: AuthenticatedRequest[AnyContent]): Fox[Task] = {
+    for {
+      annotation <- AnnotationDAO.findOneById(timeSpan.annotation.get)
+      task <- TaskService.findOneById(annotation._task.get.stringify)
+    } yield {
+      task
+    }
+  }
+
+  def totalTimeOfUser(spans: List[TimeSpan]): Long = {
+    spans.foldLeft(0l)((_, span) => span.time)
+  }
+
+  def getUsersAsStringForEmail(emails: List[String]): List[Fox[User]] = {
+    for {
+      email <- emails
```

Comment:
Same as above, you could simply write `emails.map(UserService.findOneByEmail)` and then inline this instead of having a separate method